sdk/python/kfp/cli/install_cli/kfp_installer.py

```diff
@@ -0,0 +1,191 @@
+  with open(kustomize_file, 'w') as file:
+    utils.write_line(file, 'apiVersion: kustomize.config.k8s.io/v1beta1')
+    utils.write_line(file, 'kind: Kustomization')
+    utils.write_line(file, 'bases:')
+    utils.write_line(file, '  - github.com/kubeflow/pipelines/manifests/kustomize/env/{0}{1}'.format(
+        'gcp' if enable_managed_storage else 'dev', version_ref(install_version)))
+    utils.write_line(file, '')
+    utils.write_line(file, 'commonLabels:')
+    utils.write_line(file, '  application-crd-id: kubeflow-pipelines')
+    utils.write_line(file, '')
+    utils.write_line(file, 'configMapGenerator:')
+    utils.write_line(file, '  - name: pipeline-install-config')
+    utils.write_line(file, '    env: params.env')
+    utils.write_line(file, '    behavior: merge')
+    utils.write_line(file, '')
+    utils.write_line(file, 'secretGenerator:')
+    utils.write_line(file, '  - name: mysql-secret')
+    utils.write_line(file, '    env: params-db-secret.env')
+    utils.write_line(file, '    behavior: merge')
+    utils.write_line(file, '')
+    utils.write_line(file, 'namespace: {0}'.format(namespace))
+    utils.write_line(file, '')
+
+  param_file = os.path.join(folder, 'params.env')
+  with open(param_file, 'w') as file:
+    utils.write_line(file, 'appName={0}'.format(instance_name))
+    if enable_managed_storage:
+      utils.write_line(file, 'bucketName={0}'.format(gcs_default_bucket))
+      utils.write_line(file, 'gcsProjectId={0}'.format(gcp_project_id))
+      utils.write_line(file, 'gcsCloudSqlInstanceName={0}'.format(cloud_sql_instance_name))
+
+  param_db_secret_file = os.path.join(folder, 'params-db-secret.env')
+  with open(param_db_secret_file, 'w') as file:
+    if enable_managed_storage:
+      utils.write_line(file, 'username={0}'.format(cloud_sql_username))
+      utils.write_line(file, 'password={0}'.format(cloud_sql_password))
+
+  cmd = 'kubectl apply -k {0}'.format(folder)
+  print("Executing command to apply namespace scoped resources: {0}".format(cmd))
+  result = executer.execute_subprocess(cmd)
+  if result.returncode != 0:
+    utils.print_error("Can't apply namespace scoped resources.")
+    installation_fatal(folder)
+  else:
+    utils.print_success("Successfully applied namespace scoped resources.")
+
+def wait_for_namespace_scoped_ready(instance_name, namespace):
+  cmd = 'kubectl wait applications/{0} -n {1} --for condition=Ready --timeout=1800s'.format(
+      instance_name, namespace)
+  print("Executing command to wait for namespace scoped resources ready (it may takes several mintues): {0}".format(cmd))
+  result = executer.execute_subprocess(cmd)
+  if result.returncode != 0:
+    print('Failed or timeout with code: {0}. Retry once.'.format(result.returncode))
+    result = executer.execute_subprocess(cmd)
+    if result.returncode != 0:
+      utils.print_error("Wait for namespace scoped resources be ready but failed.")
+      installation_fatal(folder)
```

Comment:
Are you sure `folder` is available here?

Reply:
thanks